apps/api/src/domain/models/shared/command-handlers/base-update-command-handler.ts

```diff
@@ -0,0 +1,46 @@
+import { ICommand } from '@coscrad/commands';
+import { Inject } from '@nestjs/common';
+import { RepositoryProvider } from '../../../../persistence/repositories/repository.provider';
+import { ResultOrError } from '../../../../types/ResultOrError';
+import { IIdManager } from '../../../interfaces/id-manager.interface';
+import { AggregateId } from '../../../types/AggregateId';
+import { ResourceType } from '../../../types/ResourceType';
+import { Resource } from '../../resource.entity';
+import { IEvent } from '../events/interfaces/event.interface';
+import { CommandHandlerBase } from './command-handler-base';
+
+export abstract class BaseUpdateCommandHandler<
+    TAggregate extends Resource
+> extends CommandHandlerBase<TAggregate> {
+    constructor(
+        protected readonly repositoryProvider: RepositoryProvider,
+        @Inject('ID_MANAGER') protected readonly idManager: IIdManager,
+        protected readonly resourceType: ResourceType
```

Comment:
As per previous comment can we remove the constructor and use DI on the `CommandHandlerBase`.

Reply:
Done